cve_bin_tool/output_engine/util.py

```diff
@@ -74,6 +81,42 @@ def format_output(all_cve_data: Dict[ProductInfo, CVEData]) -> List[Dict[str, st
     return formatted_output
 
 
+def intermediate_output(
+    all_cve_data: Dict[ProductInfo, CVEData]
+) -> Dict[Dict[str, str], List[Dict[str, str]]]:
+    """
+    summary: Generate an intermediate output in the list of dictionary format with some metadata.
+    Returns:
+        formatted_output: Dict[Dict[str, str], List[Dict[str, str]]]
+        - example:  {
+                        metadata:   {
+                                        timestamp: 2021-03-24T11:07:55Z
+                                    }
+                        report:     [
+                                        {
+                                            "vendor": "haxx"
+                                            "product": "curl",
+                                            "version": "1.2.1",
+                                            "cve_number": "CVE-1234-1234",
+                                            "severity": "LOW"
+                                        },
+                                        ...
+                                    ]
+                    }
+    """
+    formatted_output = format_output(all_cve_data)
+    intermediate_output = {}
+
+    metadata = {}
+
+    metadata["timestamp"] = datetime.now().strftime("%Y-%m-%d.%H-%M-%S")
+
+    intermediate_output["metadata"] = metadata
+    intermediate_output["report"] = formatted_output
+
+    return intermediate_output
```

Comment:
If you are not going to add any logic to this method might as well just create the dict in place:
```suggestion
    return {
        "metadata": {"timestamp": datetime.now().strftime("%Y-%m-%d.%H-%M-%S")},
        "report": format_output(all_cve_data),
    }
```